changelog.d/11950.removal

```diff
@@ -0,0 +1 @@
+Remove deprecated `user_may_create_room_with_invites` spam checker callback.
```

Comment:
It's worth noting that the initial goal of this callback was to limit what invites users could send right after creating the room. The use cases you mention are valid and can probably be implemented using [`on_create_room`](https://matrix-org.github.io/synapse/develop/modules/third_party_rules_callbacks.html#on_create_room) (which the deprecation notice doesn't mention, I'll make sure that the removal notice does). Since this callback came out of a design failure on my part, was only partly working due to `user_may_invite` (see https://github.com/matrix-org/synapse/pull/11206 for details), and seemed niche enough that there would likely not be anyone needing it, it seemed like the right thing to remove it.

> how to upgrade existing code

The expectation is that this callback was quite short-lived so there should be very little existing code. Moreover, the deprecation notice was issued back in October, which gives ample time to existing projects to come talk to us about how we can resolve things.

Reply:
Yeah, it sounds like the ultimate solution to this is to provide more details in on_create_room. It's just a shame I came into need of this mechanism only two days ago. I'll see if I can propose something.